### PR TITLE
Topic update alpaka to0.6.0rc3

### PR DESCRIPTION
**Pull request by psychocoderHPC**
Alpaka 0.6.0 has changed the namespace structure and renamed a few methods
to allocate shared memory.

- fix shadowing error shown with now alpaka:  `error: declaration of
‘size_t memsize’ shadows a parameter [-Werror=shadow]`

```
GIT_AUTHOR_NAME="Third Party" GIT_AUTHOR_EMAIL="mallocMC@hzdr.de" git subtree pull --prefix alpaka git@github.com:alpaka-group/alpaka.git release-0.6.0-rc3 --squash
```

### CODE REVIEW

File: src/include/mallocMC/creationPolicies/Scatter.hpp [hunk 1]
```diff
@@ -946,9 +946,9 @@ namespace mallocMC
                                                 "or AlignmentPolicy.");
                 }
                 auto initKernel
-                    = [] ALPAKA_FN_ACC(const AlpakaAcc& acc, T_DeviceAllocator* heap, void* heapmem, size_t memsize) {
-                          heap->pool = heapmem;
-                          heap->initDeviceFunction(acc, heapmem, memsize);
+                    = [] ALPAKA_FN_ACC(const AlpakaAcc& m_acc, T_DeviceAllocator* m_heap, void* m_heapmem, size_t m_memsize) {
+                          m_heap->pool = m_heapmem;
+                          m_heap->initDeviceFunction(m_acc, m_heapmem, m_memsize);
```
Comment (bernhardmgruber) [inline src/include/mallocMC/creationPolicies/Scatter.hpp:951]: Why is this rename necesary? and the `m_` convention seems wrong for function parameters.

File: src/include/mallocMC/creationPolicies/Scatter.hpp [hunk 2]
```diff
@@ -293,7 +293,7 @@ namespace mallocMC
                 for(;;)
                 {
                     const uint32 mask = 1 << spot;
-                    const uint32 old = alpaka::atomic::atomicOp<alpaka::atomic::op::Or>(acc, bitfield, mask);
+                    const uint32 old = alpaka::atomicOp<alpaka::AtomicOr>(acc, bitfield, mask);
```
Comment (bernhardmgruber) [inline src/include/mallocMC/creationPolicies/Scatter.hpp:296]: IIRC we have `alpaka::atomicOr(acc, bitfield, mask);` now ;)

Affects several other places as well.

**Review by bernhardmgruber — CHANGES_REQUESTED** [review]

File: src/include/mallocMC/creationPolicies/Scatter.hpp (hunk 1 above)
Comment (psychocoderHPC) [reply src/include/mallocMC/creationPolicies/Scatter.hpp:951]: Yes the compiler is showing: `error: declaration of
‘size_t memsize’ shadows a parameter [-Werror=shadow]`

File: src/include/mallocMC/creationPolicies/Scatter.hpp (hunk 2 above)
Comment (psychocoderHPC) [reply src/include/mallocMC/creationPolicies/Scatter.hpp:296]: Thx for mention it. IMO no need to change. I have not inspected the changes and used a tool for all renamings.

**Comment by psychocoderHPC** [comment]
@sbastrakov or @bernhardmgruber I need a green thump from one of you to merge it. For unknown reasons, the Travis CI is not reporting the status correctly.

**Review by sbastrakov — APPROVED** [review]

**Comment by psychocoderHPC** [comment]
reopened a clone PR in #203, Ci is buggy and therefore merging is only possible if I remove the requirements that the CI passed.

**Pull request closed without merging**